x/ecocredit/core/genesis_test.go

```diff
@@ -0,0 +1,234 @@
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+func TestValidateGenesis(t *testing.T) {
+	x := `{
+		"regen.ecocredit.v1.BatchBalance":[
+			{"address":"gydQIvR2RUi0N1RJnmgOLVSkcd4=","batch_id":"1","tradable":"90.003","retired":"9.997","escrowed":""}
+		],
+		"regen.ecocredit.v1.BatchInfo":[
+			{"issuer":"WCBEyNFP/N5RoS4h43AqkjC6zA8=","project_id":"1","batch_denom":"BIO01-00000000-00000000-001","metadata":"batch metadata","start_date":null,"end_date":null,"issuance_date":"2022-04-08T10:40:10.774108141Z"},
+		    {"issuer":null,"project_id":"1","batch_denom":"BIO02-00000000-00000000-001","metadata":"batch metadata","start_date":null,"end_date":null,"issuance_date":"2022-04-08T10:40:10.774108556Z"}
+		],
+		"regen.ecocredit.v1.BatchSequence":[{"project_id":"P01","next_batch_id":"3"}],
+		"regen.ecocredit.v1.BatchSupply":[{"batch_id":"1","tradable_amount":"90.003","retired_amount":"9.997","cancelled_amount":""}],
+		"regen.ecocredit.v1.ClassInfo":[
+			{"name":"BIO001","admin":"4A/V6LMEL2lZv9PZnkWSIDQzZM4=","metadata":"credit class metadata","credit_type":"BIO"},
+		    {"name":"BIO02","admin":"HK9YDsBMN1hU8tjfLTNy+qjbqLE=","metadata":"credit class metadata","credit_type":"BIO"}
+		],
+		"regen.ecocredit.v1.ClassIssuer":[
+			{"class_id":"1","issuer":"1ygCfmJaPVMIvVEcpx6r+2gpurM="},
+			{"class_id":"1","issuer":"KoXfzfqe+V/9x7C4XjnqDFB2Tl4="},
+			{"class_id":"2","issuer":"KoXfzfqe+V/9x7C4XjnqDFB2Tl4="},
+			{"class_id":"2","issuer":"lEjmu9Vooa24qp9vCMIlXGrMZoU="}
+		],
+		"regen.ecocredit.v1.ClassSequence":[{"credit_type":"BIO","next_class_id":"3"}],
+		"regen.ecocredit.v1.ProjectInfo":[
+			{"name":"P01","admin":"gPFuHL7Hn+uVYD6XOR00du3C/Xg=","class_id":"1","project_location":"AQ","metadata":"project metadata"},
+			{"name":"P02","admin":"CHkV2Tv6A7RXPJYTivVklbxXWP8=","class_id":"2","project_location":"AQ","metadata":"project metadata"}
+		],
+		"regen.ecocredit.v1.ProjectSequence":[{"class_id":"1","next_project_id":"3"}]}`
+
+	params := core.Params{
+		CreditTypes: []*core.CreditType{
+			{
+				Name:         "carbon",
+				Abbreviation: "C",
+				Unit:         "metric ton CO2 equivalent",
+				Precision:    6,
+			},
+			{
+				Abbreviation: "BIO",
+				Name:         "biodiversity",
+				Unit:         "ton",
+				Precision:    6,
+			},
+		},
+	}
+	core.DefaultParams()
+	err := core.ValidateGenesis(json.RawMessage(x), params)
+	require.NoError(t, err)
+}
+
+func TestGenesisValidate(t *testing.T) {
+	defaultParams := core.DefaultParams()
+
+	testCases := []struct {
+		name        string
+		gensisState func() json.RawMessage
+		params      core.Params
+		expectErr   bool
+		errorMsg    string
+	}{
+		{
+			"valid: no credit batches",
+			func() json.RawMessage {
+				return json.RawMessage(`{
+					"regen.ecocredit.v1.ClassInfo": [{"id":1,"name":"1","admin":"0lxfU2Ca/sqly8hyRhD8/lNBrvM=","metadata":"meta-data","credit_type":"C"}]
+				}`)
+			},
+			defaultParams,
+			false,
+			"",
```

Comment:
Nice to see this here. Hopefully we will have credit batches but just in case we don't.